src/Rules/Dependencies.hs

```diff
@@ -12,18 +12,21 @@ import Target
 import Utilities
 
 buildPackageDependencies :: [(Resource, Int)] -> Context -> Rules ()
-buildPackageDependencies rs context@Context {..} =
-    "//" ++ contextDir context -/- ".dependencies" %> \deps -> do
+buildPackageDependencies rs context@Context {..} = do
+    root <- buildRootRules
+    root -/- contextDir context -/- ".dependencies.mk" %> \mk -> do
         srcs <- hsSources context
         need srcs
         orderOnly =<< interpretInContext context generatedDependencies
-        let mk = deps <.> "mk"
         if null srcs
-        then writeFile' mk ""
+        then writeFileChanged mk ""
         else buildWithResources rs $
             target context (Ghc FindHsDependencies stage) srcs [mk]
         removeFile $ mk <.> "bak"
-        mkDeps <- liftIO $ readFile mk
+
+    root -/- contextDir context -/- ".dependencies" %> \deps -> do
+        need [deps <.> "mk"]
```

Comment:
Shake's `readFile'` already calls `need` on the file, so this `need` is redundant.

Reply:
Fixed.